atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/transaction/impl/SnapshotTransaction.java

```diff
@@ -1252,29 +1358,31 @@ private static boolean isSweepSentinel(Value value) {
                 } else {
                     // The value has a commit timestamp less than our start timestamp, and is visible and valid.
                     if (value.getContents().length != 0) {
-                        results.put(key, transformer.apply(value));
+                        resultsCollector.add(Maps.immutableEntry(key, transformer.apply(value)));
                         keysAddedBuilder.add(key);
                     }
                 }
             }
         }
         Set<Cell> keysAddedToResults = keysAddedBuilder.build();
-        count.addAndGet(keysAddedToResults.size());
+        resultsCounter.addAndGet(keysAddedToResults.size());
```

Comment:
resultsAddedCounter? any reason why this can't be done top level, not sure I'm following